packages/teleport/src/lib/tdp/sharedDirectoryManager.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 Gravitational, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// SharedDirectoryManager manages a FileSystemDirectoryHandle for use
+// by the TDP client. Most of it's methods can potentially throw errors
+// and so should be wrapped in try/catch blocks.
+export class SharedDirectoryManager {
+  private dir: FileSystemDirectoryHandle | undefined;
+
+  add(sharedDirectory: FileSystemDirectoryHandle) {
+    if (this.dir) {
+      throw new Error(
+        'SharedDirectoryManager currently only supports sharing a single directory'
+      );
+    }
+    this.dir = sharedDirectory;
+  }
+
+  getName(): string {
+    this.checkReady();
```

Comment:
As discussed offline these functions will be updated to use [jsdoc throws](https://jsdoc.app/tags-throws.html) and this call will add a catch.

Reply:
https://github.com/gravitational/webapps/pull/1010